kf/bpf.go

```diff
@@ -532,56 +504,64 @@ func (b *BPF) UpdateArgs(ifaceName, direction string) error {
 }
 
 // Status of user program is running
-func (b *BPF) isRunning() (bool, error) {
-	// No user program or may be disabled
+func (b *BPF) isRunning() (bool, bool, error) {
+	userProgram := true
+	bpfProgram := false
+	var err error
+
+	// CmdStatus should check for user and BPF program
 	if len(b.Program.CmdStatus) > 1 {
 		cmd := filepath.Join(b.FilePath, b.Program.CmdStatus)
 
 		if err := assertExecutable(cmd); err != nil {
-			return false, fmt.Errorf("failed to execute %s with error: %v", b.Program.CmdStatus, err)
-		}
-
-		args := make([]string, 0, len(b.Program.StatusArgs)<<1)
+			userProgram = false
+		} else {
+			args := make([]string, 0, len(b.Program.StatusArgs)<<1)
 
-		for k, val := range b.Program.StatusArgs {
-			if v, ok := val.(string); !ok {
-				err := fmt.Errorf("status args is not a string for the ebpf program %s", b.Program.Name)
-				log.Error().Err(err).Msgf("failed to convert status args value into string for program %s", b.Program.Name)
-				return false, err
+			for k, val := range b.Program.StatusArgs {
+				if v, ok := val.(string); !ok {
+					err = fmt.Errorf("status args is not a string for the ebpf program %s", b.Program.Name)
+					log.Warn().Err(err).Msgf("failed to convert status args value into string for program %s", b.Program.Name)
+				} else {
+					args = append(args, "--"+k+" ="+v)
+				}
+			}
+			prog := execCommand(cmd, args...)
+			var out bytes.Buffer
+			prog.Stdout = &out
+			prog.Stderr = &out
+			if err = prog.Run(); err != nil {
+				log.Warn().Err(err).Msgf("l3afd/nf : Failed to execute %s", b.Program.CmdStatus)
```

Comment:
What does "/nf" mean here?

```suggestion
				log.Warn().Err(err).Msgf("l3afd: Failed to execute %s", b.Program.CmdStatus)
```